stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOMStorage.java

```diff
@@ -2,20 +2,185 @@
+import com.facebook.stetho.inspector.domstorage.SharedPreferencesHelper;
 import com.facebook.stetho.inspector.jsonrpc.JsonRpcPeer;
+import com.facebook.stetho.inspector.jsonrpc.JsonRpcResult;
 import com.facebook.stetho.inspector.protocol.ChromeDevtoolsDomain;
 import com.facebook.stetho.inspector.protocol.ChromeDevtoolsMethod;
 
+import com.facebook.stetho.json.ObjectMapper;
+import com.facebook.stetho.json.annotation.JsonProperty;
+import org.json.JSONException;
 import org.json.JSONObject;
 
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
 public class DOMStorage implements ChromeDevtoolsDomain {
-  public DOMStorage() {
+  private final Context mContext;
+  private final DOMStoragePeerManager mDOMStoragePeerManager;
+  private final ObjectMapper mObjectMapper = new ObjectMapper();
+
+  public DOMStorage(Context context) {
+    mContext = context;
+    mDOMStoragePeerManager = new DOMStoragePeerManager(context);
   }
 
   @ChromeDevtoolsMethod
   public void enable(JsonRpcPeer peer, JSONObject params) {
+    mDOMStoragePeerManager.addPeer(peer);
   }
 
+  @ChromeDevtoolsMethod
   public void disable(JsonRpcPeer peer, JSONObject params) {
+    mDOMStoragePeerManager.removePeer(peer);
+  }
+
+  @ChromeDevtoolsMethod
+  public JsonRpcResult getDOMStorageItems(JsonRpcPeer peer, JSONObject params)
+      throws JSONException {
+    StorageId storage = mObjectMapper.convertValue(
+        params.getJSONObject("storageId"),
+        StorageId.class);
+
+    ArrayList<List<String>> entries = new ArrayList<List<String>>();
+    String prefTag = storage.securityOrigin;
+    if (storage.isLocalStorage) {
+      SharedPreferences prefs = mContext.getSharedPreferences(prefTag, Context.MODE_PRIVATE);
+      for (Map.Entry<String, ?> prefsEntry : prefs.getAll().entrySet()) {
+        ArrayList<String> entry = new ArrayList<String>(2);
+        entry.add(prefsEntry.getKey());
+        entry.add(SharedPreferencesHelper.valueToString(prefsEntry.getValue()));
+        entries.add(entry);
+      }
+    }
+
+    GetDOMStorageItemsResult result = new GetDOMStorageItemsResult();
+    result.entries = entries;
+    return result;
+  }
+
+  @ChromeDevtoolsMethod
+  public void setDOMStorageItem(JsonRpcPeer peer, JSONObject params) throws JSONException {
+    StorageId storage = mObjectMapper.convertValue(
+        params.getJSONObject("storageId"),
+        StorageId.class);
+    String key = params.getString("key");
+    String value = params.getString("value");
+
+    if (storage.isLocalStorage) {
+      SharedPreferences prefs = mContext.getSharedPreferences(
+          storage.securityOrigin,
+          Context.MODE_PRIVATE);
+      Object exitingValue = prefs.getAll().get(key);
+      SharedPreferences.Editor editor = prefs.edit();
+      if (!tryAssignByType(editor, key, value, exitingValue)) {
+        editor.putString(key, value);
+      }
+      editor.apply();
+    }
```

Comment:
Actually no, this is about making sure the data shows up only in the "Local Storage" group not "Session Storage".  I wanted to restrict the data so that it didn't appear to be duplicated.  Maybe we will use "Session Storage" for something else later.